### PR TITLE
feat(cloudsql): add MySQL Auto IAM AuthN sample

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Adding code sample for MySQL's automatic IAM Authentication feature that will be rolling out soon.

Blocked on [#466](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/pull/466)

### CODE REVIEW

File: cloud-sql/mysql/sqlalchemy/connect_connector_auto_iam_authn.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from google.cloud.sql.connector import Connector, IPTypes
+import pymysql
+
+import sqlalchemy
+
+
+# connect_with_connector_auto_iam_authn initializes a connection pool for
+# a Cloud SQL instance of MySQL using the Cloud SQL Python Connector
+# with Automatic IAM Database Authentication.
+def connect_with_connector_auto_iam_authn() -> sqlalchemy.engine.base.Engine:
```
Comment (dandhlee) [inline cloud-sql/mysql/sqlalchemy/connect_connector_auto_iam_authn.py:27]: Method docstrings are preferred in this format:
```suggestion
def connect_with_connector_auto_iam_authn() -> sqlalchemy.engine.base.Engine:
    """ Initializes a connection pool for a Cloud SQL instance of MySQL.
    
    Uses the Cloud SQL Python Connector with Automatic IAM Database Authentication.
    """
```

**Review by dandhlee — COMMENTED** [review]
Will you be adding tests once unblocked?

File: cloud-sql/mysql/sqlalchemy/connect_connector_auto_iam_authn.py (hunk 1 above)
Comment (jackwotherspoon) [reply cloud-sql/mysql/sqlalchemy/connect_connector_auto_iam_authn.py:27]: @dandhlee I will leave this as is for now and make a separate PR to go through and change all method comments into docstrings to have this change done in a consistent manner within this folder. That way all code snippets in the doc pages will be updated all at once.

**Review by dandhlee — APPROVED** [review]
LGTM, discussed offline that the features are tested from the client library and this sample in general (`connect` / `create_engine`) are tested.

**Pull request merged**